src/server/req_signal.c

```diff
@@ -508,12 +486,10 @@ post_signal_req(struct work_task *pwt)
 
 			svr_setjobstate(pjob, JOB_STATE_RUNNING, JOB_SUBSTATE_RUNNING);
 
-			if (!(pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob)) {
-				job_attr_def[(int) JOB_ATR_Comment].at_decode(
-						&pjob->ji_wattr[(int) JOB_ATR_Comment], NULL,
-						NULL, form_attr_comment("Job run at %s",
-						pjob->ji_wattr[(int) JOB_ATR_exec_vnode].at_val.at_str));
-			}
+			job_attr_def[(int) JOB_ATR_Comment].at_decode(
+					&pjob->ji_wattr[(int) JOB_ATR_Comment], NULL,
+					NULL, form_attr_comment("Job run at %s",
+					pjob->ji_wattr[(int) JOB_ATR_exec_vnode].at_val.at_str));
```

Comment:
Please use the above suggested function to set the comment attribute.

Reply:
fixed in rc 23 https://github.com/PBSPro/pbspro/pull/590/commits/75a2cfae3e21fc17658ef792e406c158ecdc6314